web/pandas/pdeps/00xx-string-dtype.md

```diff
@@ -0,0 +1,247 @@
+default semantics. Further, it should result in default data types when doing
+operations on the string column that result in a boolean or numeric data type
+(e.g., methods like `.str.startswith(..)` or `.str.len(..)`, or comparison
+operators like `==`, should result in default `int64` and `bool` data types).
+
+Because the original `StringDtype` implementations already use `pd.NA` and
+return masked integer and boolean arrays in operations, a new variant of the
+existing dtypes that uses `NaN` and default data types is needed.
+
+### Object-dtype "fallback" implementation
+
+To avoid a hard dependency on PyArrow for pandas 3.0, this PDEP proposes to keep
+a "fallback" option in case PyArrow is not installed. The original `StringDtype`
+backed by a numpy object-dtype array of Python strings can be used for this, and
+only need minor updates to follow the above-mentioned missing value semantics
+([GH-58451](https://github.com/pandas-dev/pandas/pull/58451)).
+
+For pandas 3.0, this is the most realistic option given this implementation is
+already available for a long time. Beyond 3.0, we can still explore further
+improvements such as using nanoarrow or NumPy 2.0, but at that point that is an
+implementation detail that should not have a direct impact on users (except for
+performance).
+
+### Naming
+
+Given the long history of this topic, the naming of the dtypes is a difficult
+topic.
+
+In the first place, we need to acknowledge that most users should not need to
+use storage-specific options. Users are expected to specify `pd.StringDtype()`
+or `"string"`, and that will give them their default string dtype (which
+depends on whether PyArrow is installed or not).
+
+But for testing purposes and advanced use cases that want control over this, we
+need some way to specify this and distinguish them from the other string dtypes.
+Currently, the `StringDtype(storage="pyarrow_numpy")` is used, where
+"pyarrow_numpy" is a rather confusing option.
+
+TODO see if we can come up with a better naming scheme
+
+## Alternatives
+
+### Why not delay introducing a default string dtype?
+
+To avoid introducing a new string dtype while other discussions and changes are
+in flux (eventually making pyarrow a required dependency? adopting `pd.NA` as
+the default missing value sentinel? using the new NumPy 2.0 capabilities?), we
+could also delay introducing a default string dtype until there is more clarity
+in those other discussions.
+
+However:
+
+1. Delaying has a cost: it further postpones introducing a dedicated string
+   dtype that has massive benefits for our users, both in usability as (for the
+   significant part of the user base that has PyArrow installed) in performance.
+2. In case we eventually transition to use `pd.NA` as the default missing value
+   sentinel, we will need a migration path for _all_ our data types, and thus
+   the challenges around this will not be unique to the string dtype and
+   therefore not a reason to delay this.
```

Comment:
other than maybe this is the only default data type that will be using PyArrow (I mention elsewhere that maybe we need to discuss the consequences of this)

Reply:
> other than maybe this is the only default data type that will be using PyArrow

Which will actually make it _easier_ to transition to use `pd.NA`, since we already have a variant of the dtype that does that?

Reply:
Yeah. This text is fine as is.